processors/elasticsearch-cfenv-processor/src/main/java/com/ibm/cfenv/spring/boot/elasticsearch/ElasticsearchCfEnvProcessor.java

```diff
@@ -0,0 +1,75 @@
+    private static final String REST_CLIENT = "rest";
+    
+    @Override
+    public boolean accept(CfService service) {
+        return service.existsByLabelStartsWith("databases-for-elasticsearch");
+    }
+
+    @Override
+    public CfEnvProcessorProperties getProperties() {
+        return CfEnvProcessorProperties.builder()
+                .propertyPrefixes("sslcontext,elasticsearch")
+                .serviceName("elasticsearch")
+                .build();
+    }
+
+    @Override
+    public void process(CfCredentials cfCredentials, Map<String, Object> properties) {
+        List<String> clients = getClients(System.getenv("IBM_CFENVPROCESSOR_ELASTICSEARCH_CLIENTS_ENABLE"));
+        
+        Map<String, Object> credentialsData = cfCredentials.getMap();
+        Map<String, Object> connection = (Map<String, Object>) credentialsData.get("connection");
+        Map<String, Object> https = (Map<String, Object>) connection.get("https");
+        List<Map<String, Object>> hosts = (List<Map<String, Object>>) https.get("hosts");
+        Map<String, Object> host = hosts.get(0);
+        String hostname = (String) host.get("hostname");
+        int port = (int) host.get("port");
+        
+        
+        Map<String, String> authentication = (Map<String, String>) https.get("authentication");        
+        Map<String, String> certificate = (Map<String, String>) https.get("certificate");
+        String username = authentication.get("username");
+        String password = authentication.get("password");
+        String uris = "https://" + hostname + ":" + port;
+        
+        if (clients.contains(REST_CLIENT)) {
+            properties.put("spring.elasticsearch.rest.uris", uris);
+            properties.put("spring.elasticsearch.rest.username", username);
+            properties.put("spring.elasticsearch.rest.password", password);
+        }
+
+        if (clients.contains(REACTIVE_CLIENT)) {
+            properties.put("spring.data.elasticsearch.client.reactive.endpoints", hostname + ":" + port);
+            properties.put("spring.data.elasticsearch.client.reactive.username", username);
+            properties.put("spring.data.elasticsearch.client.reactive.password", password);
+        } else {
+            properties.put("spring.elasticsearch.rest.uris", uris);
+            properties.put("spring.elasticsearch.rest.username", username);
+            properties.put("spring.elasticsearch.rest.password", password);
+        }
+        
+        properties.put("sslcontext.contexts.elasticsearch.certificate", certificate.get("certificate_base64"));
+    }
+    
+    private List<String> getClients(String elasticSearchClients) {
+        if (elasticSearchClients == null || elasticSearchClients.isEmpty()) {
+            return new ArrayList<String>();
+        }
+        return Arrays.asList(System.getenv("IBM_CFENVPROCESSOR_ELASTICSEARCH_CLIENTS_ENABLE").toLowerCase().split(","));
+    }
```

Comment:
Normalise the return from getClients to contain the default if the env var was empty.. 
Then the logic in the main section (l50-l63) can be simplified to just 2 if blocks, rather than if, if/else. 
Today if the user specs rest&reactive, you'll do the work in l50-l54 and then also do l60-l64, which could lead to issues in future if those two code blocks ever fall out of sync with one another, better to avoid the situation entirely.